python/setup_cluster/ssh_client_handler.py

```diff
@@ -0,0 +1,115 @@
+import paramiko
+import copy
+
+
+class SSHClientHandler:
+    def __init__(
+        self,
+        node_ssh_username,
+        web_ingress_ssh_username,
+        web_ingress_public_ip,
+        web_ingress_private_ip,
+        logger,
+    ):
+        self.node_ssh_username = node_ssh_username
+        self.web_ingress_ssh_username = web_ingress_ssh_username
+        self.web_ingress_public_ip = web_ingress_public_ip
+        self.web_ingress_private_ip = web_ingress_private_ip
+        self.logger = logger
+
+    def create_ssh_client(self, ip, username, use_jump=False):
+        self.logger.debug(f"func: create_ssh_client {ip=} {username=} {use_jump=}")
+        try:
+            jumpbox_channel = None
+            if use_jump:
+                jumpbox = paramiko.SSHClient()
+                jumpbox.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+                jumpbox.connect(
+                    self.web_ingress_public_ip, username=self.web_ingress_ssh_username
+                )
+                jumpbox_transport = jumpbox.get_transport()
+                src_addr = (self.web_ingress_private_ip, 22)
+                dest_addr = (ip, 22)
+                jumpbox_channel = jumpbox_transport.open_channel(
+                    "direct-tcpip", dest_addr, src_addr
+                )
+            client = paramiko.SSHClient()
+            client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            client.connect(
+                ip,
+                username=username,
+                sock=jumpbox_channel,
+                banner_timeout=30,
+                timeout=30,
+            )
+            return client
+        except Exception as e:
+            self.logger.error(f"Failed to create SSH client for {ip}: {e}")
+            return None
+
+    def execute_commands(self, commands, ip, use_jump=False, run_sequenctially=False):
+        self.logger.debug("func: execute_commands")
+        proxy = None
+        try:
+            ssh_client = self.create_ssh_client(ip, self.node_ssh_username, use_jump)
+            if ssh_client:
+                output = ""
+                if run_sequenctially:
```

Comment:
Just curious, what's the functional difference between run_sequentially=True and run_sequentially=False? Why use one over the other?